frontend/src/hooks/api/kms/mutations.tsx

```diff
@@ -96,3 +101,37 @@ export const useLoadProjectKmsBackup = (projectId: string) => {
     }
   });
 };
+
+export const useExternalKmsFetchGcpKeys = (orgId: string) => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async ({
+      gcpRegion,
+      ...rest
+    }: Pick<ExternalKmsGcpSchemaType, "gcpRegion"> &
+      (
+        | (Pick<ExternalKmsGcpSchemaType, "credential"> & { kmsId?: never })
+        | { kmsId: string; credential?: never }
+      )): Promise<{ keys: string[] }> => {
+      const { credential, kmsId } = rest;
+
+      if ((credential && kmsId) || (!credential && !kmsId)) {
+        throw new Error("Either 'credential' or 'kmsId' must be provided, but not both.");
+      }
+
+      const apiUrl = kmsId
+        ? `/api/v1/external-kms/fetch-gcp-keys/${kmsId}`
+        : "/api/v1/external-kms/fetch-gcp-keys/credential";
+
+      const { data } = await apiRequest.post(apiUrl, {
+        region: gcpRegion,
+        ...rest
+      });
+
+      return data;
+    },
```

Comment:
why do we use two different routes for these and not just one?

Reply:
Yeah I think this will cause confusion since the function name remains the same